src/core/util.ts

```diff
@@ -682,31 +732,36 @@ export class HashMap<T, KEY extends string | number = string | number> {
     // (We usually treat `null` and `undefined` as the same, different
     // from ES6 Map).
     get(key: KEY): T {
-        return this.data.hasOwnProperty(key) ? this.data[key] : null;
+        return this.data.get(key);
     }
     set(key: KEY, value: T): T {
         // Comparing with invocation chaining, `return value` is more commonly
         // used in this case: `const someVal = map.set('a', genVal());`
-        return (this.data[key] = value);
+        this.data.set(key, value);
+        return value;
     }
     // Although util.each can be performed on this hashMap directly, user
     // should not use the exposed keys, who are prefixed.
     each<Context>(
         cb: (this: Context, value?: T, key?: KEY) => void,
         context?: Context
     ) {
-        for (let key in this.data) {
-            if (this.data.hasOwnProperty(key)) {
-                cb.call(context, this.data[key], key);
-            }
-        }
+        this.data.forEach((key, value) => {
+            cb.call(context, value, key);
+        });
     }
     keys(): KEY[] {
-        return keys(this.data);
+        // Polyfilled map return and Array<KEYS> for keys
+        if (this.data instanceof MapPolyfill) {
```

Comment:
Maybe we can use the  `isNativeMapSupported` constant instead of the `instanceof` operator to detect if it's the map polyfill. (Not sure)

Reply:
I think we can, but it exposes a possible problem if the user or some 3rd party package is applying polyfills after this module is evaluated. I would tend to say that while possible, it is probably very low risk. We should be able to assume that any polyfills should still correctly implement the native Map interface.